syntax = "proto3";

package gatherly;


service Gatherly {
  // query a single hashed_kmer
  rpc gather_kmer (KmerHash) returns (KmerSources) {}

  // query a full sourmash sig (gzipped or not)
  // TODO use stream here
  rpc gather_sig (sourmashSig) returns (Sources) {}

}

// query a single hashed_kmer
message KmerHash {
    int64 kmer_hash = 1;
}

message KmerSources {
  string sources = 1;
}

// query a complete sourmash signature
message sourmashSig{
  string sig_path = 1;
  int64 kSize = 2;
}

message Sources {
  string sources = 1;
}
